fastly/block_fastly_service_v1_s3logging.go

```diff
@@ -0,0 +1,313 @@
+package fastly
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+var s3loggingSchema = &schema.Schema{
+	Type:     schema.TypeSet,
+	Optional: true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			// Required fields
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Unique name to refer to this logging setup",
+			},
+			"bucket_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "S3 Bucket name to store logs in",
+			},
+			"s3_access_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("FASTLY_S3_ACCESS_KEY", ""),
+				Description: "AWS Access Key",
+				Sensitive:   true,
+			},
+			"s3_secret_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("FASTLY_S3_SECRET_KEY", ""),
+				Description: "AWS Secret Key",
+				Sensitive:   true,
+			},
+			// Optional fields
+			"path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Path to store the files. Must end with a trailing slash",
```

Comment:
We should probably add a validation function to this. (Not sure how the API responds without it.)